### PR TITLE
STJ: Use base getter as fallback solution

### LINKED ISSUE #96996: System.Text.Json ignored properties that override the setter only.
A bit of a rare corner case, but the serializer isn't currently handling properties that only override the setter:
```C#
using System.Text.Json;

string json = JsonSerializer.Serialize(new Derived { X = 42 });
Console.WriteLine(json); // {}

public class Base
{
    public virtual int X { get; set; }
}

public class Derived : Base
{
    public override int X { set => base.X = value; }
}
```
This is because the serializer only inspects the signature of the override, but it should instead try to honor the signature of the base definition. Impacts both reflection and source generated serializers.

### PR DESCRIPTION
**Pull request by manandre**
Fixes #96996

### CODE REVIEW

File: src/libraries/System.Text.Json/tests/System.Text.Json.Tests/NewtonsoftTests/JsonSerializerTests.cs [hunk 1]
```diff
@@ -157,4 +163,14 @@ public class ChildClass : BaseClass
     {
         public new bool IsTransient { get; set; }
     }
+
+    public class Base
```
Comment (eiriktsarpalis) [inline src/libraries/System.Text.Json/tests/System.Text.Json.Tests/NewtonsoftTests/JsonSerializerTests.cs:167]: Thanks! Would it be possible to add source generator tests as well?

Comment (manandre) [reply src/libraries/System.Text.Json/tests/System.Text.Json.Tests/NewtonsoftTests/JsonSerializerTests.cs]: Test rewrritten as a `PropertyVisibility` generic test, run for both Reflection and Source Generator use cases.
Unfortunately, the test fails in the SrcGen case. I would need some help here. The getter cannot be set afterwards.

Comment (eiriktsarpalis) [reply src/libraries/System.Text.Json/tests/System.Text.Json.Tests/NewtonsoftTests/JsonSerializerTests.cs]: Yeah, that is probably expected. There is a corresponding method in the source generator that should be synced:

https://github.com/dotnet/runtime/blob/3ae99a78d8967e5f19765c52f272febef5def623/src/libraries/System.Text.Json/gen/JsonSourceGenerator.Parser.cs#L1000-L1051

Comment (manandre) [reply src/libraries/System.Text.Json/tests/System.Text.Json.Tests/NewtonsoftTests/JsonSerializerTests.cs]: Method updated accordingly, but does not generate the code correctly. It seems that I modify the derived property too late...

Comment (eiriktsarpalis) [reply src/libraries/System.Text.Json/tests/System.Text.Json.Tests/NewtonsoftTests/JsonSerializerTests.cs]: >  It seems that I modify the derived property too late...

Can you clarify what you mean by that? An example of a failing test and what the error is would help.

**Review by jeffhandley — CHANGES_REQUESTED** [review]
There are still failing tests that need to be resolved.

**Pull request closed without merging**